jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbTransientDiagnosticsCollector.java

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+* Copyright (c) 2022 IBM Corporation, Adit Rada and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation, Adit Rada - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4jakarta.jdt.core.jsonb;
+
+import java.util.List;
+
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.ISourceRange;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4jakarta.jdt.core.DiagnosticsCollector;
+import org.eclipse.lsp4jakarta.jdt.core.JDTUtils;
+import org.eclipse.lsp4jakarta.jdt.core.JakartaCorePlugin;
+
+public class JsonbTransientDiagnosticsCollector implements DiagnosticsCollector {
```

Comment:
Rather than creating an entire new diagnostics collector class and duplicating some logic, can we add this logic to the existing `JsonbCreatorDiagnosticsCollector` class and consider renaming to `JsonbDiagnosticsCollector`?

Reply:
Done